### DISCUSSION TITLE
tutorials/voyage-part3/05-local-schema-checks

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#75] [course feedback]
# tutorials/voyage-part3/05-local-schema-checks

https://www.apollographql.com/tutorials/voyage-part3/05-local-schema-checks

**Comment by elgizabbasov** [comment]
I have a best practices question, why did we decide to use @shareable directive for GalacticCoordinates instead of making it an interface? Where can I read more on these decisions & best practices for using entity vs interface vs @shareable? Thank you for the amazing tutorials!

**Reply by elgizabbasov** [reply]
Furthermore, when to use Unions instead of Interface, when to use Enums instead of Interface and etc

**Reply by mabuyo** [reply]
Hey there @elgizabbasov ! Great questions. 

We have this resource on [**Recommended usage for GraphQL interfaces**](https://www.apollographql.com/docs/technotes/TN0017-interfaces/).

You can also read up on [**Thinking in entities**](https://www.apollographql.com/docs/technotes/TN0026-thinking-in-entities/).

And we have [**a short explainer on enums here**](https://graphql.com/learn/enums/#enums).

I hope these help, feel free to follow up if not!

**Comment by sunnytoolsmiths** [comment]
what is the difference using @key and @shareable? both identify a type that can be resued cross multiple subgraphs, right?

**Reply by sunnytoolsmiths** [reply]
nvm, I found it out!

**Comment by bhavikji** [comment]
Couple of links are broken in this doc viz;, [Voyage II](https://www.apollographql.com/tutorials/voyage-part2/value-types) and [covered the @shareable directive in our discussion on field sharing](https://www.apollographql.com/tutorials/voyage-part2/08-publishing-our-schemas)
